app/views/devise/registrations/_new.html.erb

```diff
@@ -28,13 +28,27 @@
   <% end %>
 
   <fieldset>
-    <%= f.text_field :password, autocomplete: "off", id: "signup_password", placeholder: "Your Password *" %>
+    <%= f.password_field :password, autocomplete: "off", id: "signup_password", placeholder: "Your Password *" %>
   </fieldset>
 
   <fieldset>
-    <%= f.text_field :password_confirmation, autocomplete: "off", placeholder: "Re-enter Your Password *" %>
+    <%= f.password_field :password_confirmation, autocomplete: "off", placeholder: "Re-enter Your Password *" %>
+  </fieldset>
+
+  <fieldset>
+    <%= check_box_tag('hide_signup_pass', false, true) %>
+    <%= label_tag('hide_signup_pass', 'Show password', class: 'plain') %>
   </fieldset>
 
   <%= f.submit "Sign Up" %>
 
 <% end %>
+
+<script type="text/javascript">
```

Comment:
Stylistically, I'd rather see this in it's own js file because that's where I would go to look for it.